crates/networking/rpc/engine/fork_choice.rs

```diff
@@ -112,34 +114,25 @@ impl RpcHandler for ForkChoiceUpdatedV3 {
 
         // Build block from received payload. This step is skipped if applying the fork choice state failed
         let mut response = ForkChoiceResponse::from(PayloadStatus::valid_with_hash(
-            self.fork_choice_state.head_block_hash,
+            request.fork_choice_state.head_block_hash,
         ));
 
-        match &self.payload_attributes {
+        match &request.payload_attributes {
             // Payload may be invalid but we had to apply fork choice state nevertheless.
             Err(e) => return Err(RpcErr::InvalidPayloadAttributes(e.into())),
             Ok(None) => (),
             Ok(Some(attributes)) => {
                 info!("Fork choice updated includes payload attributes. Creating a new payload.");
-                let chain_config = context.storage.get_chain_config()?;
-                if !chain_config.is_cancun_activated(attributes.timestamp) {
-                    return Err(RpcErr::UnsuportedFork(
-                        "forkChoiceV3 used to build pre-Cancun payload".to_string(),
-                    ));
-                }
-                if attributes.timestamp <= head_block.timestamp {
-                    return Err(RpcErr::InvalidPayloadAttributes(
-                        "invalid timestamp".to_string(),
-                    ));
-                }
+                let chain_config = storage.get_chain_config()?;
+                fork_choice_updated.validate(attributes, chain_config, head_block)?;
```

Comment:
So this is the only thing that is different between the FCU calls? What if there is any actions besides validating?

Reply:
For this call the only difference are the validations and the `version` field in the `PayloadArgs` [here](https://github.com/lambdaclass/ethrex/pull/1297/files#diff-4af4eb82669a3403cab38afef8fb3b766f9ded917858c00d0f9c905fde5bcb53R135). If some other action or validation is required we can extend the `ForkChoiceUpdatedImpl` trait to handle it in each version that implements it